influxql/functions.gen.go

```diff
@@ -28,19 +28,59 @@ func NewFloatFuncReducer(fn FloatReduceFunc) *FloatFuncReducer {
 	return &FloatFuncReducer{fn: fn}
 }
 
-func (r *FloatFuncReducer) Aggregate(p *FloatPoint) {
-	t, v, aux := r.fn(r.prev, p)
-	if r.prev == nil {
-		r.prev = &FloatPoint{}
+func (r *FloatFuncReducer) Aggregate(points ...*FloatPoint) {
+	for _, p := range points {
+		t, v, aux := r.fn(r.prev, p)
+		if r.prev == nil {
+			r.prev = &FloatPoint{}
+		}
+		r.prev.Time = t
+		r.prev.Value = v
+		r.prev.Aux = aux
+		if p.Aggregated > 1 {
+			r.prev.Aggregated += p.Aggregated
+		} else {
+			r.prev.Aggregated++
+		}
 	}
-	r.prev.Time = t
-	r.prev.Value = v
-	r.prev.Aux = aux
-	r.prev.Aggregated++
 }
 
-func (r *FloatFuncReducer) Emit() *FloatPoint {
-	return r.prev
+func (r *FloatFuncReducer) Emit() []FloatPoint {
+	return []FloatPoint{*r.prev}
+}
+
+// FloatReduceSliceFunc is the function called by a FloatPoint reducer.
+type FloatReduceSliceFunc func(a []FloatPoint) []FloatPoint
+
+type FloatSliceFuncReducer struct {
+	points []FloatPoint
+	fn     FloatReduceSliceFunc
+}
+
+func NewFloatSliceFuncReducer(fn FloatReduceSliceFunc) *FloatSliceFuncReducer {
+	return &FloatSliceFuncReducer{fn: fn}
+}
+
+func (r *FloatSliceFuncReducer) Aggregate(points ...*FloatPoint) {
```

Comment:
Why are we using []*FloatPoint here since it makes this implementation require more allocs?

Seems like using []FloatPoint would work fine. Am I missing something?